src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -629,9 +632,20 @@ export const Deposit: React.FC<DepositProps> = ({
     ],
   )
 
-  const validateFiatAmount = useCallback(
+  const validateCryptoAmount = useCallback(
+    (value: string) => {
+      contextDispatch?.({ type: ThorchainSaversDepositActionType.SET_LOADING, payload: true })
+      return _validateCryptoAmount(value).finally(() => {
```

Comment:
.finally() vs .then() so we properly fall the promise through (`.finally()` doesn't change the value of the awaited promise expression, vs .then and .catch doing so)

Reply:
I personally find it much cleaner and declarative to have the loading setters nice and around the promise, but happy to keep `validateCryptoAmount` if that makes things looker nicer

Reply:
ab5b41412d for the early return on `!dispatch`